packages/adminrouter/extra/src/master/cache.lua

```diff
@@ -293,6 +294,82 @@ local function fetch_and_store_state_mesos(auth_token)
 end
 
 
+local function fetch_and_store_mesos_leader_state()
+    -- Lua forbids jumping over local variables definition, hence we define all
+    -- of them here.
+    local res_body, r, err, answers
+
+    if HOST_IP == 'unknown' then
+        ngx.log(ngx.ERR,
+			"Local Mesos Master IP address is unknown, cache entry is unusable")
+        res_body = '{"is_local": "unknown", "leader_ip": null}'
+        goto store_cache
+    end
+
+    -- We want to use Navstar's dual-dispatch approach and get the response
+    -- as fast as possible from any operational MesosDNS. If we set it to local
+    -- instance, its failure will break this part of the cache as well.
+	--
+	-- Also, see https://github.com/openresty/lua-resty-dns#limitations
+    r, err = resolver:new{
+        nameservers = {{"198.51.100.1", 53},
+                       {"198.51.100.2", 53},
+                       {"198.51.100.3", 53}},
+        retrans = 3,  -- 5 retransmissions on receive timeout
```

Comment:
I have changed it to:
```
retrans = 3,  -- retransmissions on receive timeout
timeout = 2000,  -- msec
```

So that we do not have to keep comments in sync. Thx!